ccmlib/common.py

```diff
@@ -76,6 +77,36 @@ class UnavailableSocketError(CCMError):
     pass
 
 
+class LogPatternToVersion(object):
+    def __init__(self, patterns_to_versions):
+        assert type(patterns_to_versions) is dict, "LogPatternToVersion takes a dictionary."
+        self.versions_to_patterns = patterns_to_versions
+
+    def __call__(self, version):
+        pattern = None
+        for key in self.versions_to_patterns.keys():
```

Comment:
I don't think this will work correctly, as you can't depend on iteration order to be anything in particular.